Xamarin.Forms.Core/Registrar.cs

```diff
@@ -21,6 +21,9 @@ public class Registrar<TRegistrable> where TRegistrable : class
 
 		public void Register(Type tview, Type trender)
 		{
+			//avoid caching null renderers
```

Comment:
We _do_ want to cache null renderer results if they're valid, so we don't have to do multiple `GetCustomAttribute` calls (see the comments in `GetHandlerType`.

It looks like what you really want is to avoid overwriting an existing handler with `null`; something like this:
```
if (trender == null && _handlers.ContainsKey(tview)) 
    return;
```

So if a handler has already been registered, don't overwrite it with 'null'; if there's no handler, we can cache the `null` result.